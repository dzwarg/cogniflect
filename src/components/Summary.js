import React from 'react';
import {SmartComponent as StartOver} from './StartOver';
import {Col, Grid, Jumbotron, Row} from 'react-bootstrap';
import { connect } from 'react-redux';

const fmt = (val) => (`${(100 * val).toFixed(2)} %`);
const Summary = ({individualCorrect, teamCorrect, totalQuestions}) => (
  <Grid>
    <Row>
      <Col>
        <Jumbotron>
          <h1>Quiz Summary</h1>
          <p>
            Research has demonstrated that people perform better as teams for these types of tasks. Typical results are 15% accuracy for individuals, and 100% accuracy for groups. How did you fare?
          </p>
        </Jumbotron>
      </Col>
    </Row>
    <Row>
      <Col xs={12} md={6}>
        <h2>Individually</h2>
        <h3>Score: {fmt(individualCorrect/totalQuestions)}</h3>
        <p>You answered {individualCorrect} out of {totalQuestions} correctly.</p>
      </Col>
      <Col xs={12} md={6}>
        <h2>Collaboratively</h2>
        <h3>Score: {fmt(teamCorrect/totalQuestions)}</h3>
        <p>Your team answered {teamCorrect} out of {totalQuestions} correctly.</p>
      </Col>
    </Row>
    <StartOver />
    <Row>
      <Col xs={12}>
        <h2>References</h2>
        <ul>
          <li><a href="http://journal.sjdm.org/15/151029/jdm151029.html">Investigating an alternate form of the cognitive reflection test</a> (<a href="http://journal.sjdm.org/vol11.1.html">Judgment and Decision Making, Vol. 11, No. 1</a>)</li>
          <li><a href="https://en.wikipedia.org/wiki/Cognitive_reflection_test">Cognitive Reflection Test</a> (Wikipedia)</li>
        </ul>
      </Col>
    </Row>
  </Grid>
);

const mapStateToProps = (state) => ({
  totalQuestions: state.appState.get('questions').size,
  individualCorrect: state.appState.get('questions').filter((val) => (
    val.get('myAnswer') === val.get('truth')
  )).size,
  teamCorrect: state.appState.get('questions').filter((val) => (
    val.get('ourAnswer') === val.get('truth')
  )).size
});

export const DumbComponent = Summary;
export const SmartComponent = connect(mapStateToProps)(Summary);