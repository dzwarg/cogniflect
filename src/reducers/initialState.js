import {fromJS} from 'immutable';

const initialState = fromJS({
  teamCode: '',
  userTeamCode: '',
  assessmentType: 'individual',
  questionCursor: 0,
  questions: [
    {
      id: 0,
      text: "In a lake, there is a patch of lily pads. Every day, the patch doubles in size. If it takes 48 days for the patch to cover the entire lake, how long would it take for the patch to cover half of the lake?",
      truth: 1,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: '24' },
        { id: 1, text: '47' },
        { id: 2, text: '96' },
        { id: 3, text: '49' }
      ],
      next: 1
    },
    {
      id: 1,
      text: "A bat and a ball cost $1.10 in total. The bat costs $1.00 more than the ball. How much does the ball cost?",
      truth: 0,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: '$0.05' },
        { id: 1, text: '$0.06' },
        { id: 2, text: '$0.10' },
        { id: 3, text: '$0.50' }
      ],
      next: 2
    },
    {
      id: 2,
      text: "If it takes 5 machines 5 minutes to make 5 widgets, how long would it take 100 machines to make 100 widgets?",
      truth: 0,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: '5 minutes' },
        { id: 1, text: '100 minutes' },
        { id: 2, text: '500 minutes' },
        { id: 3, text: '10000 minutes' }
      ],
      next: 3
    },
    {
      id: 3,
      text: "If you’re running a race and you pass the person in second place, what place are you in?",
      truth: 1,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: '1st place' },
        { id: 1, text: '2nd place' },
        { id: 2, text: '3rd place' },
        { id: 3, text: 'Last place' }
      ],
      next: 4
    },
    {
      id: 4,
      text: "A farmer had 15 sheep and all but 8 died. How many are left?",
      truth: 2,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: '6' },
        { id: 1, text: '7' },
        { id: 2, text: '8' },
        { id: 3, text: '9' }
      ],
      next: 5
    },
    {
      id: 5,
      text: "Emily’s father has three daughters. The first two are named April and May. What is the third daughter’s name?",
      truth: 3,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: 'June' },
        { id: 1, text: 'August' },
        { id: 2, text: 'Summer' },
        { id: 3, text: 'Emily' }
      ],
      next: 6
    },
    {
      id: 6,
      text: "How many cubic feet of dirt are there in a hole that is 3’ deep x 3’ wide x 3’ long?",
      truth: 0,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: '0 ft^3' },
        { id: 1, text: '3 ft^3' },
        { id: 2, text: '9 ft^3' },
        { id: 3, text: '27 ft^3' }
      ],
      next: null
    }
  ],
  team: {
    members: 0,
    synced: 0
  }
});

export default initialState;