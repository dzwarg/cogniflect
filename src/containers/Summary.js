import { connect } from 'react-redux'
import Summary from '../components/Summary'

const mapStateToProps = (state) => ({
  totalQuestions: state.appState.get('questions').size,
  individualCorrect: state.appState.get('questions').filter((val) => (
    val.get('myAnswer') === val.get('truth')
  )).size,
  teamCorrect: state.appState.get('questions').filter((val) => (
    val.get('ourAnswer') === val.get('truth')
  )).size
})

export default connect(mapStateToProps)(Summary)
export {mapStateToProps}