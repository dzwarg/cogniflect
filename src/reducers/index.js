import { combineReducers } from 'redux'
import initialState from './initialState'; 

const appState = (state = initialState, action) => {
  switch (action.type) {
    case 'SET':
      return state.setIn(action.payload.path, action.payload.value);
    default:
      return state
  }
}

export default combineReducers({
  appState
});