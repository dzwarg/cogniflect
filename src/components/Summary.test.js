import React from 'react';
import {SmartComponent as Summary, DumbComponent} from './Summary';
import { createStore } from 'redux';
import reducers from '../reducers';
import {fromJS} from 'immutable';
import ShallowRenderer from 'react-test-renderer/shallow';

const initialState = {
  appState: fromJS({
    questions: [{
      myAnswer: true,
      ourAnswer: false,
      truth: true,
    },{
      myAnswer: false,
      ourAnswer: true,
      truth: true
    }]    
  })
};

const store = createStore(reducers, initialState);
  
describe('smart component', () => {
  var renderedComp;
  beforeEach(() => {
    const renderer = new ShallowRenderer();
    renderedComp = renderer.render(
      <Summary store={store}/>
    );
  });

  it('matches snapshot', () => {
    expect(renderedComp).toMatchSnapshot()
  });
});

describe('dumb component', () => {
  var renderedComp;
  beforeEach(() => {
    const renderer = new ShallowRenderer();
    renderedComp = renderer.render(
      <DumbComponent {...initialState}/>
    );
  });
  
  it('matches snapshot', () => {
    expect(renderedComp).toMatchSnapshot();
  });
});

