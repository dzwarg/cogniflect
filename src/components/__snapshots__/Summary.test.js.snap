// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dumb component matches snapshot 1`] = `
<Grid
  bsClass="container"
  componentClass="div"
  fluid={false}
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
    >
      <Jumbotron
        bsClass="jumbotron"
        componentClass="div"
      >
        <h1>
          Quiz Summary
        </h1>
        <p>
          Research has demonstrated that people perform better as teams for these types of tasks. Typical results are 15% accuracy for individuals, and 100% accuracy for groups. How did you fare?
        </p>
      </Jumbotron>
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={6}
      xs={12}
    >
      <h2>
        Individually
      </h2>
      <h3>
        Score: 
        NaN %
      </h3>
      <p>
        You answered 
         out of 
         correctly.
      </p>
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      md={6}
      xs={12}
    >
      <h2>
        Collaboratively
      </h2>
      <h3>
        Score: 
        NaN %
      </h3>
      <p>
        Your team answered 
         out of 
         correctly.
      </p>
    </Col>
  </Row>
  <Connect(StartOver) />
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      xs={12}
    >
      <h2>
        References
      </h2>
      <ul>
        <li>
          <a
            href="http://journal.sjdm.org/15/151029/jdm151029.html"
          >
            Investigating an alternate form of the cognitive reflection test
          </a>
           (
          <a
            href="http://journal.sjdm.org/vol11.1.html"
          >
            Judgment and Decision Making, Vol. 11, No. 1
          </a>
          )
        </li>
        <li>
          <a
            href="https://en.wikipedia.org/wiki/Cognitive_reflection_test"
          >
            Cognitive Reflection Test
          </a>
           (Wikipedia)
        </li>
      </ul>
    </Col>
  </Row>
</Grid>
`;

exports[`smart component matches snapshot 1`] = `
<Summary
  dispatch={[Function]}
  individualCorrect={1}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": null,
    }
  }
  teamCorrect={1}
  totalQuestions={2}
/>
`;
